app/src/main/java/com/github/giommok/softwaredevproject/FirebaseAccount.java

```diff
@@ -50,6 +53,7 @@ public void onDataChange(@NonNull DataSnapshot snapshot) {
             if(snapshot.hasChild(Database.CHILD_USERNAME))
                 syncGetUsernameFromProfile(snapshot.child(Database.CHILD_USERNAME).getValue(String.class));
             else username = "null";
+            syncFriendsFromProfile(snapshot.child(Database.CHILD_FRIENDS));
```

Comment:
Check if _Firends_ child exists for the current user profile. If it doesn't exist and you try to call the sync method, an Exception will be thrown and the rest of the sync methods would not be called...
**TODO** : check if the  `if(snapshot.hasChild(Database.CHILD_FRIENDS))` is correct since `CHILD_FRIENDS` contains two `"/"`
```suggestion
if(snapshot.hasChild(Database.CHILD_FRIENDS))
            syncFriendsFromProfile(snapshot.child(Database.CHILD_FRIENDS));
else friends = new ArrayList<FriendItem>();
```

Reply:
Discussed on Discord.